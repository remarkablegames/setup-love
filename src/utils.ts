import { platform } from 'node:os';

/**
 * Gets download object.
 *
 * @see {@link https://github.com/love2d/love/releases}
 *
 * @param version - LÃ–VE version
 * @returns - Download object
 */
export function getDownloadObject(version: string) {
  let binaryDirectory = '';
  let filename = 'love';
  let url = `https://github.com/love2d/love/releases/download/${version}/love-${version}-`;

  switch (platform()) {
    case 'darwin':
      binaryDirectory = 'love.app/Contents/MacOS';
      url += 'macos.zip';
      break;

    case 'linux':
      binaryDirectory = 'squashfs-root/bin';
      url += 'x86_64.AppImage';
      break;

    case 'win32':
      binaryDirectory = `love-${version}-win64`;
      filename += '.exe';
      url += 'win64.zip';
      break;
  }

  return {
    binaryDirectory,
    filename,
    url,
  };
}
