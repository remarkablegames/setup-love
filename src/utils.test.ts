import os from 'os';

import { getDownloadObject } from './utils';

jest.mock('os');
const mockedOs = jest.mocked(os);

const platforms: NodeJS.Platform[] = ['darwin', 'linux', 'win32'];

describe('getDownloadObject', () => {
  describe.each(platforms)('when platform is %p', (os) => {
    const version = '11.4';

    beforeEach(() => {
      jest.resetAllMocks();
      mockedOs.platform.mockReturnValueOnce(os);
    });

    it('gets download object', () => {
      expect(getDownloadObject(version)).toMatchSnapshot();
    });
  });
});
