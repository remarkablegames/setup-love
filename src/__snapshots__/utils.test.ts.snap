// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`getDownloadObject when platform is "darwin" gets download object 1`] = `
{
  "binaryDirectory": "love.app/Contents/MacOS",
  "filename": "love",
  "url": "https://github.com/love2d/love/releases/download/11.4/love-11.4-macos.zip",
}
`;

exports[`getDownloadObject when platform is "linux" gets download object 1`] = `
{
  "binaryDirectory": "squashfs-root/bin",
  "filename": "love",
  "url": "https://github.com/love2d/love/releases/download/11.4/love-11.4-x86_64.AppImage",
}
`;

exports[`getDownloadObject when platform is "win32" gets download object 1`] = `
{
  "binaryDirectory": "love-11.4-win64",
  "filename": "love.exe",
  "url": "https://github.com/love2d/love/releases/download/11.4/love-11.4-win64.zip",
}
`;
